OnboardPietroPieri/App.tsx

```diff
@@ -5,23 +5,138 @@
  * @format
  */
 
-import React from "react";
-import {View, Text, StyleSheet} from 'react-native';
+import React, {PropsWithChildren, useState} from "react";
+import {
+  Alert,
+  TouchableOpacity,
+  TextInput,
+  SafeAreaView,
+  View,
+  Text,
+  StyleSheet
+} from 'react-native';
 
 const styles = StyleSheet.create({
+  safeArea: {
+    marginTop: 80,
+    flex: 1,
+  },
   center: {
+    marginTop: 30,
     alignItems: 'center',
-    flex: 1,
     justifyContent: 'center',
+    flex: 0.3,
+    paddingHorizontal: 24,
+  },
+  header: {
+    fontSize: 30,
+    fontWeight: 'bold',
+    textAlign: 'center',
+    color: 'black',
+  },
+  inputText: {
+    color: 'black',
+    fontWeight: 'normal',
+    marginLeft: 40,
+    top: 28,
+  },
+  input: {
+    marginLeft: 40,
+    marginTop: 30,
+    height: 50,
+    borderColor: 'gray',
+    borderWidth: 2,
+    paddingHorizontal: 10,
+    width: '80%',
+    borderRadius: 10,
   },
+  button: {
+    height: 50,
+    marginTop: 50,
+    marginLeft: 40,
+    backgroundColor: '#9403fc',
+    padding: 10,
+    borderRadius: 10,
+    width: '80%',
+  },
+  buttonText: {
+    textAlign: 'center',
+    fontWeight: 'bold',
+    color: 'white',
+    fontSize: 16,
+  }
+  
+
 });
 
-const App = () => {
+type SectionProps = PropsWithChildren<{
+  type?: string,
+  title: string;
+  setLogin: React.Dispatch<React.SetStateAction<string>>;
+}>;
+
+
+
+function Section({title, setLogin, type}: SectionProps): React.JSX.Element {
+  const isPassword = type === 'password';
+
   return (
-    <View style={styles.center}>
-      <Text>"Hello word"</Text>
+    <View>
+      <Text style={styles.inputText}>{title}</Text>
+      <TextInput
+        style={styles.input}
+        keyboardType={isPassword ? 'default' : 'email-address'}
+        autoCapitalize="none"
+        secureTextEntry={isPassword}
+        onChangeText={(text: React.SetStateAction<string>) => setLogin(text)}>
+      </TextInput>
     </View>
   );
+}
+
+const App = () => {
+
+  const [email, setEmail] = useState('');
+  const [password, setPassword] = useState('');
+
+  const isValidEmail = (email: string) => {
+    const emailRegex = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;
+    return emailRegex.test(email);
+  };
+
+  const isValidPassword = (password: string) => {
+    return password.length >= 7 && /[a-zA-Z]/.test(password) && /\d/.test(password);
+  };
```

Comment:
uma dica interessante é que as funções criadas dentro do componentes vão ser recriadas sempre que houver renderização, mesmo que não seja necessário, em alguns casos isso pode afetar a quantidade de renderizações e prejudicar a performance. Pra esses casos tem algumas alternativas, se a função for independente do componente ela pode ser declarada fora dele e basicamente vai ser a mesma sempre que for chamada, no outro caso se tiver alguma dependencia vc pode utilizar hooks pra gerenciar quando a função dever ou não ser alterada, normalmente é utilizado o `useCallback`.

No seu caso pode simplesmente colocar essas funções pra fora do seu componente App sem problemas

Reply:
Se eu colocar a função para fora gero esse erro:
<img width="1464" alt="Screenshot 2024-01-08 at 13 49 47" src="https://github.com/indigotech/onboard-pietro-pieri/assets/69023934/09d996d5-dcab-4165-96cc-82715cd07a27">

como posso resolver isso.

Reply:
Acredito que você tenha aplicado as duas sugestões diferentes. Seriam duas opções:
a) se a função for independente do componente ela pode ser declarada fora dele e basicamente vai ser a mesma sempre que for chamada
b) no outro caso se tiver alguma dependencia vc pode utilizar hooks pra gerenciar quando a função dever ou não ser alterada, normalmente é utilizado o `useCallback`

Reply:
Não consegui resolver esse problema podemos conversar depois